File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        List<int[]> array = new ArrayList<>();

        for(int i = 0; i < nums.length; i++){
            map.put(nums[i], map.getOrDefault(nums[i], 0) + 1);
        }

        for(Map.Entry<Integer, Integer> entry: map.entrySet()){
            array.add(new int[]{entry.getValue(), entry.getKey()});
        }

        array.sort((a,b) -> b[0] - a[0]);
        int[] result = new int[k];
        for(int i = 0; i < k; i++){
            result[i] = array.get(i)[1];
        }
        return result;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-4.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        i = 0
        sort = sorted(nums)
        result = []
        for i in range(len(sort)):
            j = i + 1
            k = len(nums) - 1
            if(i > 0 and sort[i] == sort[i - 1]):
                continue
            else:
                while(j < k):
                    if(sort[j] + sort[k] + sort[i] > 0):
                        k -= 1
                    elif(sort[j] + sort[k] + sort[i] < 0):
                        j += 1
                    elif(sort[j] + sort[k] + sort[i] == 0):
                        result.append([sort[i], sort[j], sort[k]])
                        j += 1
                        k -= 1
                        while j < k and sort[j] == sort[j - 1]:
                            j += 1
                        while j < k and sort[k] == sort[k + 1]:
                            k -= 1
                    else:
                        continue
        return result